From 664692beeda1da0ecf038976a71914096727c7ca Mon Sep 17 00:00:00 2001
From: Philippe Coval <p.coval@samsung.com>
Date: Mon, 18 Jun 2018 11:53:43 +0200
Subject: [PATCH] tizenrt: Support build in docker

Usage:
  docker build -f tests/port/tizenrt/Dockerfile .

This will help to reproduce failures on reference env

Bug: https://jira.iotivity.org/projects/LITE/issues/LITE-4
Change-Id: I3b230c31c2657a6b30b459900627a19e707a4a02
Signed-off-by: Philippe Coval <p.coval@samsung.com>
---
 tests/port/tizenrt/Dockerfile | 62 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 62 insertions(+)
 create mode 100755 tests/port/tizenrt/Dockerfile

diff --git a/tests/port/tizenrt/Dockerfile b/tests/port/tizenrt/Dockerfile
new file mode 100755
index 00000000..7b612d7c
--- /dev/null
+++ b/tests/port/tizenrt/Dockerfile
@@ -0,0 +1,62 @@
+#!/bin/echo docker build . -f
+# -*- coding: utf-8 -*-
+#{
+# Copyright 2018 Samsung Electronics France SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#}
+
+FROM debian:stable
+MAINTAINER Philippe Coval (philippe.coval@osg.samsung.com)
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV LC_ALL en_US.UTF-8
+ENV LANG ${LC_ALL}
+
+RUN echo "#log: Configuring locales" \
+  && set -x \
+  && apt-get update -y \
+  && apt-get install -y locales \
+  && echo "${LC_ALL} UTF-8" | tee /etc/locale.gen \
+  && locale-gen ${LC_ALL} \
+  && dpkg-reconfigure locales \
+  && sync
+
+RUN echo "#log: ${project}: Setup system" \
+  && set -x \
+  && apt-get update -y \
+  && apt-get install -y \
+  build-essential \
+  curl \
+  git \
+  make \
+  python \
+\
+  gawk \
+  && apt-get clean \
+  && sync
+
+ENV project iotivity-constrained
+
+ADD . /usr/local/src/${project}/${project}/
+WORKDIR /usr/local/src/${project}/${project}/
+RUN echo "#log: ${project}: Preparing sources" \
+  && set -x \
+  && sync
+
+RUN echo "#log: ${project}: Building sources" \
+  && set -x \
+  && cd tests \
+  && ls \
+  &&  ./auto_build.sh tizenrt \
+  && sync
-- 
2.16.1.windows.1

